shell/platform/darwin/ios/framework/Source/FlutterPlatformViewsTest.mm

```diff
@@ -143,4 +144,25 @@ - (void)testCanCreatePlatformViewWithoutFlutterView {
   flutterPlatformViewsController->Reset();
 }
 
+// TODO(justinmc): I wrote this test before realizing handleMethodCall is
```

Comment:
You might be able to extend the class and expose a public version in the test class?

Reply:
You should be able to make the channel with the same identifier and send a message to it and wait for a reply.  There should be an example of this somewhere in our tests but I can't find one.  Did you look at this test for ideas?  https://github.com/gaaclarke/engine/blob/linting-warning/shell/platform/darwin/common/framework/Source/FlutterChannelsTest.m

Reply:
I tried this:
```objc
- (void)testClipboard {
  id project = OCMClassMock([FlutterDartProject class]);
  FlutterEngine* engine = [[FlutterEngine alloc] initWithName:@"foobar" project:project];
  FlutterMethodChannel* channel =
      [FlutterMethodChannel methodChannelWithName:@"flutter/platform"
                                  binaryMessenger:engine.binaryMessenger
                                            codec:[FlutterJSONMethodCodec sharedInstance]];
  [engine run];
  XCTestExpectation* didCallReply = [self expectationWithDescription:@"didCallReply"];
  [channel invokeMethod:@"Clipboard.getData"
           arguments:nil
              result:^(id _Nullable result){
                [didCallReply fulfill];
              }];
  [self waitForExpectationsWithTimeout:1.0 handler:nil];
}
```

But it crashes because it tries to run on a fake project.  Channels don't work unless an engine is running and the engine doesn't run unless there is a project and in unit tests the engine doesn't have a project...   I'll look into this a bit more this afternoon to see what the best approach will be.

Reply:
Thanks I appreciate the help.

Reply:
Ok, I looked at a few options and this appears to be more complicated than I had hoped.  [`handleMethodCall` isn't private](https://github.com/gaaclarke/engine/blob/master/shell/platform/darwin/ios/framework/Source/FlutterPlatformPlugin.h#L16:L16) so you can just call it directly for the test.

Reply:
Ah I forgot I could just call it with square brackets in objective c.  That seems to work.  Thanks for the help!